src/app.ts

```diff
@@ -266,7 +274,7 @@ export default class App extends Vue {
                 // eslint-disable-next-line
                 const { hash, sender, value, data, height } = tx;
                 const vote = parseVote(data, config.type);
-                if (vote.name === config.name) {
+                if (vote.name === config.name && isValidChoices(vote.choices)) {
```

Comment:
Good thought! Though technically, you can add whatever "invalid" choice, they'll just be ignored. The validation so far is simply if parsing the vote succeeds, that's a valid vote. Done. I'd leave it at that. If that validation should be part, then I'd suggest to extend parseVote(data, config) to do that test. The test is equal to something like `choices.filter((choice) => !valid.includes(choice.name)).length > 0`, right?

Reply:
But just to clarify: I'd not add that.

Reply:
I have added the check at this place, so that for invalid votes the balance does not need to be requested from the network in the first place.